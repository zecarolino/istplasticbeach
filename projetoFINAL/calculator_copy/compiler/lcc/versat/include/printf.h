#ifndef _PRINTF_H_
#define _PRINTF_H_

#include "printi.h"
#include "puts.h"
#include "stdarg.h"

int printf(const char *fmt, ...)
{
  va_list ap;
  va_start(ap, fmt);
  while (*fmt != 0) {
    if (*fmt == '%') {
      fmt++;
      if (*fmt == 'd') printi(va_arg(ap,int), 10);
      else if (*fmt == 's') puts(va_arg(ap,char*));
      else if (*fmt == 'c') putchar(va_arg(ap,int));
      else putchar(*fmt);
    }
    else putchar(*fmt);
    fmt++;
  }
  va_end(ap);
  return 1;
}

#endif /* _PRINTF_H_ */
