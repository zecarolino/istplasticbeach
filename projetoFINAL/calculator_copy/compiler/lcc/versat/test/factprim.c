#include "printf.h"

int nextPrime(int n)
{
  int i;

  if (n <= 0) return 0;
  if (n <= 2) return n+1;
  while (++n > 0) {
    if (n % 2 == 0) continue; /* is even */
    for (i = 3; i < n; i += 2) /* test odd numbers */
      if (n % i == 0)
	break;
    if (i >= n)
      return n;
  }
  return 0; /* overflow */
}

void factprim(int num)
{
  int cnt, n = 1;

  while (n < num) {
    n = nextPrime(n);
    for (cnt = 0; num % n == 0; cnt++)
      num /= n;
    if (cnt != 0)
      printf(" * %d^%d", n, cnt);
  }
  printf("\n");
}

int main()
{
  factprim(144); /* expect ' * 2^4 * 3^2' */
  return 0;
}
