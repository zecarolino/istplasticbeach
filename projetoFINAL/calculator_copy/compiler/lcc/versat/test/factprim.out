VCD info: dumpfile xtop.vcd opened for output.
 * 2^4 * 3^1
0
Catched Trap at time  15445 clock cycles:
